server/src/actors/project.rs

```diff
@@ -44,13 +45,29 @@ pub enum ProjectError {
 
 impl ResponseError for ProjectError {}
 
+#[derive(Clone, Debug, Eq, PartialEq, Hash)]
+pub enum RetryAfterScope {
+    Key(String),
+    Project,
+}
+
+impl RetryAfterScope {
+    fn get_variants(meta: &EventMeta) -> impl Iterator<Item = RetryAfterScope> {
```

Comment:
As per Nomicon, please name this `iter_variants` or just `variants`. `get_` is not to be used according to Rust naming conventions.

_nit_: This can be implemented less obscured using `std::iter::once`.